html_css/v2/main-course/custom-properties.md

```diff
@@ -0,0 +1,205 @@
+
+### Learning Outcomes
+
+* You'll learn how to declare a custom property
+* You'll learn how to access a custom property in a rule declaration
+
+### Using Custom Properties
+
+The syntax for declaring a custom property is really simple, and it's actually pretty similar to how we make a normal rule declaration:
+
+~~~css
+.error-modal {
+  --color-error-text: red;
+}
+~~~
+
+That's it! First, we start with a double hyphen followed by a case-sensitive, hyphen-separated property name (`color-error-text` wouldn't be the same as `Color-Error-Text`). The use of single hyphens to separate words is very important here because spaces are not valid (`--color error text` would not work).
+
+Then we can store any valid CSS value in our newly declared custom property. Yes, *any valid CSS value*. So while we can use custom properties for something as simple as a color, we can also use them to store slightly more complex or even shorthand values:
+
+~~~css
+.error-modal {
+  --modal-border: 1px solid black;
+  --main-font: calc(2rem + 5vw);
+}
+~~~
+
+So now we know how to declare a custom property, but how do we actually *access* it in a rule declaration? As simply as declaring it, of course:
+
+~~~css
+.error-modal {
+  /* declaring the custom properties */
+  --color-error-text: red;
+  --main-font: 1.25rem;
+  --modal-border: 1px solid black;
+
+  /* accessing those custom properties */
+  border: var(--modal-border);
+  color: var(--color-error-text);
+  font-size: var(--main-font);
+}
+~~~
+
+We use the `var()` function as the value of a CSS property, and then place our custom property inside of the parenthesis (including the double hyphen at the beginning).
+
+### Scope
+
+In the last example above, you may have noticed that we declared and then accessed our custom properties within the same declaration block. That's because the scope of a custom property is determined by the selector.
+
+~~~html
+<p class='cool-paragraph'>Lorem ipsum dolor sit amet.</p>
+
+<p class='boring-paragraph'>Lorem ipsum dolor sit amet.</p>
+~~~
+~~~css
+.cool-paragraph {
+  --main-bg: red;
+
+  background-color: var(--main-bg);
+}
+
+.boring-paragraph {
+  background-color: var(--main-bg);
+}
+~~~
+
+In the above example, only the element with the `cool-paragraph` class gets styled with a red background, while the `boring-paragraph` selector doesn't since it's not within the scope of where the custom property was declared. If you're familiar with how scope works in JavaScript, this sort of behavior should feel a little similar.
+
+That doesn't mean a custom property can *only* be accessed on the selector it was declared for, though. The scope includes the selector the custom property was declared for as well as the selector's descendants. Let's take a look at a modified version of the above example:
+
```

Comment:
I think it could help to explain the example before demonstrating it. This way we can reduce people being confused when trying to follow along expecting this to work, just to tell them in the next paragraph it won't work.

It's alright to have supporting text under the example, but it may not be a good idea to put an example that a person may expect to work before telling them it's not going to work.